src/plugins/safeSnap/utils/umaModule.ts

```diff
@@ -263,3 +357,188 @@ export const getModuleDetailsUma = async (
+          [',', 'explanation', ':', '"']
+        ),
+        toUtf8Bytes(explanation.replace(/^0x/, '')),
+        pack(
+          ['string', 'string', 'string', 'string', 'string'],
+          ['"', ',', 'rules', ':', '"']
+        ),
+        toUtf8Bytes(rules.replace(/^0x/, '')),
+        pack(['string'], ['"'])
+      ]
+    );
+  } else {
+    return {
+      dao: moduleDetails[0][0],
+      oracle: moduleDetails[1][0],
+      rules: moduleDetails[2][0],
+      minimumBond: minimumBond,
+      expiration: moduleDetails[4][0],
+      allowance: bondDetails.currentUserBondAllowance,
+      collateral: bondDetails.collateral,
+      decimals: bondDetails.decimals,
+      symbol: bondDetails.symbol,
+      userBalance: bondDetails.currentUserBalance,
+      needsBondApproval: needsApproval,
+      noTransactions: true,
+      activeProposal: false,
+      assertionEvent: undefined,
+      proposalExecuted: false,
+      livenessPeriod: livenessPeriod
+    };
+  }
+  // Check for active proposals. proposal hash can be identical across assertions
+  // but the explanation field should be unique. we will filter this out later.
+  const assertionId = await moduleContract.assertionIds(proposalHash);
+
+  const activeProposal =
+    assertionId !==
+    '0x0000000000000000000000000000000000000000000000000000000000000000';
+
+  // Search for requests with matching ancillary data
+  const oracleContract = new Contract(
+    optimisticOracle,
+    UMA_ORACLE_ABI,
+    provider
+  );
+
+  const latestBlock = await provider.getBlock('latest');
+  // modify this per chain. this should be updated with constants for all chains. start block is og deploy block.
+  // this needs to be optimized to reduce loading time, currently takes a long time to parse 3k blocks at a time.
+  const oGstartBlock = network === '1' ? 17167414 : 0;
+  const maxRange = network === '1' ? 3000 : 10000;
+
+  // TODO: use og subgraph to replace need for these calls
+  const [transactionsProposedEvents, executionEvents] = await Promise.all([
+    // Check if this specific proposal has already been executed.
+    // note usage of pageEvents, which query only based on a limit number of blocks within a broader range
+    // this prevents block range too large errors.
+    pageEvents(
+      oGstartBlock,
+      latestBlock.number,
+      maxRange,
+      ({ start, end }: { start: number; end: number }) => {
+        return moduleContract.queryFilter(
+          moduleContract.filters.TransactionsProposed(),
+          start,
+          end
+        );
+      }
+    ),
+    pageEvents(
+      oGstartBlock,
+      latestBlock.number,
+      maxRange,
+      ({ start, end }: { start: number; end: number }) => {
+        return moduleContract.queryFilter(
+          moduleContract.filters.ProposalExecuted(proposalHash),
+          start,
+          end
+        );
+      }
+    )
+  ]);
+  const [assertion0] = await findAssertionsGql(network, {
```

Comment:
this is where we have removed previous event based queries and replaced with subgraph, and mapped the data identically to how we were doing it before